src/components/CadenceEditor.tsx

```diff
@@ -33,49 +33,45 @@ class CadenceEditor extends React.Component<{
 
     this.editorStates = {};
 
-    if (typeof document !== "undefined") {
-      this.handleResize = this.handleResize.bind(this);
-      window.addEventListener("resize", this.handleResize);
-      // NOTE: monaco is browser-only, pre-render of the app is done via node;
-      // Check if document exists to be sure we're in browser land
-      // before loading monaco.
-      this.monaco = require("monaco-editor");
-      configureMonaco(this.monaco);
-    }
+    this.handleResize = this.handleResize.bind(this);
+    window.addEventListener("resize", this.handleResize);
+    // NOTE: monaco is browser-only, pre-render of the app is done via node;
+    // Check if document exists to be sure we're in browser land
+    // before loading monaco.
+    this.monaco = require("monaco-editor");
+    configureMonaco(this.monaco);
   }
 
   handleResize() {
     this.editor && this.editor.layout();
   }
 
   componentDidMount() {
-    if (typeof document !== "undefined") {
-      this.monaco = require("monaco-editor");
-      configureMonaco(this.monaco);
-      const monacoOptions = {
-        language: "Cadence",
-        minimap: {
-          enabled: false
-        }
-      };
-
-      this.editor = this.monaco.editor.create(
-        document.getElementById(this.props.mount),
-        monacoOptions
-      );
-
-      this._subscription = this.editor.onDidChangeModelContent((event: any) => {
-        const code = this.editor.getValue().replace(/\r\n/g, '\n')
-        this.props.onChange(code, event);
-      });
-
-      const state = this.getOrCreateEditorState(
-        this.props.activeId,
-        this.props.code
-      );
-      this.editor.setModel(state.model);
-      this.editor.focus();
-    }
+    this.monaco = require("monaco-editor");
+    configureMonaco(this.monaco);
```

Comment:
Hmm, I just noticed that this code is called in both the constructor and `componentDidMount`. Maybe it can be removed from the constructor?

Reply:
Good catch! I don't know why that was done and if it is necessary. I've only removed the now unnecessary `if (typeof document !== "undefined") {` wrappers in this PR, but I'll have a look into this after the language server PR is merged